env/composite_env_wrapper.h

```diff
@@ -727,11 +746,24 @@ class LegacyWritableFileWrapper : public FSWritableFile {
                   IODebugContext* /*dbg*/) override {
     return status_to_io_status(target_->Append(data));
   }
+  IOStatus AppendWithVerify(
```

Comment:
Why add the new API rather than expand the IOOptions to have a DataVerificationInfo element

Reply:
IOOptions suppose to bring the hints down to the FS, might not be a appropriate place to bring the verification information.